lib/analyzer/docker-analyzer.ts

```diff
@@ -19,14 +19,12 @@ export async function analyze(
   dockerfileAnalysis?: dockerFile.DockerFileAnalysis,
   options?: DockerOptions,
 ) {
-  const [imageInspection, osRelease] = await Promise.all([
-    imageInspector.detect(targetImage, options),
-    osReleaseDetector.detectDynamically(
-      targetImage,
-      dockerfileAnalysis,
-      options,
-    ),
-  ]);
+  const imageInspection = await imageInspector.detect(targetImage, options);
+  const osRelease = await osReleaseDetector.detectDynamically(
+    targetImage,
+    dockerfileAnalysis,
+    options,
+  );
```

Comment:
Removed the promise.all to make sure that imageInspector throws first. osReleaseDetector runs `docker run` which will throw if the image couldn't be pulled. For the sake of throwing an error that reflects that we tried pulling for inspect, changed this.

Reply:
It might make sense to add a comment here. This is some important knowledge which is not obvious from the code. Just to prevent someone refactoring the code again into a Promise.all :smile: